src/Illuminate/Foundation/Testing/DatabaseTransactions.php

```diff
@@ -4,12 +4,42 @@
 
 trait DatabaseTransactions
 {
+    /**
+     * The connections currently affected by the transaction. Indexed by connection name.
+     *
+     * @var array
+     */
+    protected $transactingConnections = [];
+
+    /**
+     * Retrieve the names of the database connections that should be included in the transaction.
+     *
+     * The default value of empty string will cause only the default database connection to be included in the transaction.
+     *
+     * @return array
+     */
+    protected function connectionsToTransact()
+    {
+        return [''];
+    }
+
     public function beginDatabaseTransaction()
     {
-        $this->app->make('db')->beginTransaction();
+        $db = $this->app->make('db');
+
+        foreach ($this->connectionsToTransact() as $connectionName) {
+            $connection = $db->connection($connectionName);
+            $this->transactingConnections[$connectionName] = $connection;
+
+            $connection->beginTransaction();
+        }
 
         $this->beforeApplicationDestroyed(function () {
-            $this->app->make('db')->rollBack();
+
+            foreach ($this->connectionsToTransact() as $connectionName) {
+                $this->app->make('db')->connection($connectionName)->rollBack();
+            }
+
```

Comment:
Friend, I appreciate the work you do for open source but you're *really* not making me want to help out **at all**.

What code style? The contribution guide only says to run php-cs-fixer, which I have. I follow PSR-2. Are you complaining about the space above and below the foreach? Since I'm supposed to "look at every other file in the framework" I looked at `Illuminate\Support\Collection` which spaces every foreach like this. Is that not what you want now?

You can either have a contribution guide that details what you actually want, or you can tell people what you want in the pulls, but when you do *neither* you just come off as a jerk.